### PR TITLE
Database Setup

### PR DESCRIPTION
**Pull request by mandy132002**

### CODE REVIEW

File: app/src/main/java/com/gdsctsec/smartt/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -2,16 +2,20 @@ package com.gdsctsec.smartt.ui.main
...
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
 import androidx.core.view.get
 import androidx.viewpager2.widget.ViewPager2
 import com.gdsctsec.smartt.R
 import com.gdsctsec.smartt.ui.main.adapter.ViewPagerAdapter
 import com.google.android.material.bottomnavigation.BottomNavigationView
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
```
Comment (SirusCodes) [inline app/src/main/java/com/gdsctsec/smartt/ui/main/MainActivity.kt]: Revert this file

**Review by SirusCodes — CHANGES_REQUESTED** [review]

**Review by SirusCodes — APPROVED** [review]

File: app/src/main/java/com/gdsctsec/smartt/database/LectureDao.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+interface LectureDao {
+
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    suspend fun saveLecture(lec: TimeTable)
+
+    @Update
+    suspend fun updateLecture(lec: TimeTable)
+
+    @Query("SELECT * FROM timeTable WHERE weekday = :weekday")
+    suspend fun getLectureByWeekday(weekday: Weekday): List<TimeTable>
```
Comment (2tanayk) [inline app/src/main/java/com/gdsctsec/smartt/database/LectureDao.kt]: this is for displaying on the home screen right?then wrap with `LiveData`, like `LiveData<List<TimeTable>>`

File: app/src/main/java/com/gdsctsec/smartt/database/Database.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@Database(entities = [TimeTable::class], version = 1, exportSchema = false)
+abstract class LectureDatabase : RoomDatabase(){
+
+    abstract fun lectureDao(): LectureDao
+
+    companion object{
+        @Volatile
+        private var INSTANCE: LectureDatabase? = null
+
+        fun getInstance(context: Context): LectureDatabase {
```
Comment (2tanayk) [inline app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: from where will this `getInstance()` be called?

**Review by 2tanayk — CHANGES_REQUESTED** [review]
All is fine, just some changes but, what about other entities?

File: app/src/main/java/com/gdsctsec/smartt/database/LectureDao.kt (hunk 2 above)
Comment (SirusCodes) [reply app/src/main/java/com/gdsctsec/smartt/database/LectureDao.kt]: Cool can do that

File: app/src/main/java/com/gdsctsec/smartt/database/Database.kt (hunk 3 above)
Comment (SirusCodes) [reply app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: I think it will be in VM

Comment (2tanayk) [reply app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: if you want the database to be created as soon as the database object is first created, then here is the code:
```
 companion object {
        @Volatile
        private var instance: AppDatabase? = null
        private val LOCK = Any()

        operator fun invoke(context: Context) = instance ?: synchronized(LOCK) {
            instance ?: buildDatabase(context).also {
                instance = it
            }
        }

        private fun buildDatabase(context: Context) =
            Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "database.db"
            ).build()
    }
```

File: app/src/main/java/com/gdsctsec/smartt/database/Database.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package com.gdsctsec.smartt.database
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+
+@Database(entities = [TimeTable::class], version = 1, exportSchema = false)
+abstract class LectureDatabase : RoomDatabase(){
```
Comment (2tanayk) [inline app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: can you rename this as `AppDatabase`?, that sounds intuitive

**Review by 2tanayk — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/gdsctsec/smartt/database/Database.kt (hunk 3 above)
Comment (SirusCodes) [reply app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: The current code does the same, just syntax is different

File: app/src/main/java/com/gdsctsec/smartt/database/Database.kt (hunk 4 above)
Comment (SirusCodes) [reply app/src/main/java/com/gdsctsec/smartt/database/Database.kt]: Ok cool

**Pull request merged**